app/helpers/stream_entries_helper.rb

```diff
@@ -48,22 +48,25 @@ def microformats_h_class(status, is_predecessor, is_successor, include_threads)
   end
 
   def rtl?(text)
-    rtl_characters = /[\p{Hebrew}|\p{Arabic}|\p{Syriac}|\p{Thaana}|\p{Nko}]+/m.match(text)
+    rtl_charcount = text.gsub(/[^\p{Hebrew}|\p{Arabic}|\p{Syriac}|\p{Thaana}|\p{Nko}]/m, '').strip.size.to_f
 
-    if rtl_characters.present?
-      total_size = text.strip.size.to_f
-      rtl_size(rtl_characters.to_a) / total_size > 0.3
+    if rtl_charcount.positive?
+      # Remove mentions before counting characters to decide RTL ratio
+      justtext = text.gsub(Account::MENTION_RE, '')
+      # Remove Email addresses
+      justtext = justtext.gsub(/^([a-z0-9_\.-]+)@([\da-z\.-]+)\.([a-z\.]{2,6})$/, '')
+      # Naiive catcher for URLs
+      justtext = justtext.gsub(/[-A-Za-z0-9@:%._\+~#=]{2,256}\.[a-z]{2,6}\b([-A-Za-z0-9@:%_\+.~#?&\/=]*)/, '')
```

Comment:
I fear detection of the language may be more computationally expensive than three gsub() calls, but I don't intend to dig deeper into that 3rd party code. If it ran once on writing to the DB and written for future reads it would make some sense, but as it is, we have to detect directionality on the way out to the browser (or in the browser when talking JS), and I patched the code that does it today to do it better, that's all. maybe in the stretch till v1.5 I'll have time to test or improve the language detection mechanism.